client/task-list/tasks/payments/paypal.js

```diff
@@ -1,118 +1,151 @@
+		>
+			{ __( 'Connect', 'woocommerce-admin' ) }
+		</a>
+	);
+}
+
+class PayPal extends Component {
 	constructor( props ) {
 		super( props );
 
 		this.state = {
 			autoConnectFailed: false,
 			connectURL: '',
-			isPending: false,
 		};
 
-		this.updateSettings = this.updateSettings.bind( this );
+		this.enablePaypalPlugin = this.enablePaypalPlugin.bind( this );
+		this.updateManualSettings = this.updateManualSettings.bind( this );
 		this.validate = this.validate.bind( this );
 	}
 
 	componentDidMount() {
-		const { createNotice, markConfigured } = this.props;
-
+		const { createNotice } = this.props;
 		const query = getQuery();
 		// Handle redirect back from PayPal
-		if ( query[ 'paypal-connect' ] ) {
-			if ( query[ 'paypal-connect' ] === '1' ) {
-				createNotice(
-					'success',
-					__( 'PayPal connected successfully.', 'woocommerce-admin' )
-				);
-				markConfigured( 'paypal' );
+		if ( query.onboarding ) {
+			if (
+				query.onboarding === 'complete' &&
+				! query[ 'ppcp-onboarding-error' ]
+			) {
+				this.enablePaypalPlugin();
 				return;
 			}
 
-			/* eslint-disable react/no-did-mount-set-state */
-			this.setState( {
-				autoConnectFailed: true,
-			} );
-			/* eslint-enable react/no-did-mount-set-state */
+			if ( query[ 'ppcp-onboarding-error' ] ) {
+				/* eslint-disable react/no-did-mount-set-state */
+				this.setState( {
+					autoConnectFailed: true,
+				} );
+				createNotice(
+					'error',
+					__(
+						'There was a problem saving your payment settings through the onboarding, please fill them in manually.',
+						'woocommerce-admin'
+					)
+				);
+			}
 			return;
 		}
-
-		this.fetchOAuthConnectURL();
+		this.fetchOAuthConnectURLAndOnboardingSetup();
 	}
 
 	componentDidUpdate( prevProps ) {
 		const { activePlugins } = this.props;
 
 		if (
-			! prevProps.activePlugins.includes(
-				'woocommerce-gateway-paypal-express-checkout'
-			) &&
-			activePlugins.includes(
-				'woocommerce-gateway-paypal-express-checkout'
-			)
+			! prevProps.activePlugins.includes( PAYPAL_PLUGIN ) &&
+			activePlugins.includes( PAYPAL_PLUGIN )
 		) {
-			this.fetchOAuthConnectURL();
+			this.fetchOAuthConnectURLAndOnboardingSetup();
 		}
 	}
 
-	isWooCommerceServicesConnected() {
-		const {
-			activePlugins,
-			isJetpackConnected,
-			wcsTosAccepted,
-		} = this.props;
-
-		return (
-			isJetpackConnected &&
-			wcsTosAccepted &&
-			activePlugins.includes( 'woocommerce-services' )
-		);
-	}
-
-	async fetchOAuthConnectURL() {
+	async fetchOAuthConnectURLAndOnboardingSetup() {
 		const { activePlugins } = this.props;
 
-		if (
-			! activePlugins.includes(
-				'woocommerce-gateway-paypal-express-checkout'
-			)
-		) {
+		if ( ! activePlugins.includes( PAYPAL_PLUGIN ) ) {
 			return;
 		}
 
 		this.setState( { isPending: true } );
 		try {
 			const result = await apiFetch( {
-				path: WC_ADMIN_NAMESPACE + '/plugins/connect-paypal',
+				path: WC_PAYPAL_NAMESPACE + '/onboarding/get-params',
```

Comment:
Getting the information from the paypal payments plugin now instead.
If this `404` meaning outdated, it will show the manual set up.

Reply:
This also shows the manual setup if a `500` is returned, not sure if it should show an error instead.

Reply:
It should show the manual input fields in that case, but probably also show an error.